File: dynamic-programming/LC-53/bruteforce.java
class Solution {
    public int maxSubArray(int[] nums) {
        // Time Complexity: O(n^2)
        // Space Complexity: O(1)
        int maximum = Integer.MIN_VALUE;

        for (int i = 0; i < nums.length; i++) {

            int sum = 0;

            for (int j = i; j < nums.length; j++) {

                sum += nums[j];

                if (sum > maximum) {
                    maximum = sum;
                }
            }
        }

        return maximum;
    }
}

File: dailystreak/LC-1674(13thMay26)/bruteforce.py
#gives tle dont try
##time complexity: O(n * m) where n is the length of nums
# and m is the range of possible sums (2*limit)  
#space complexity: O(1) as we are using constant space


class Solution:
    def minMoves(self, nums: List[int], limit: int) -> int:
        n = len(nums)
        ans = float('inf')

        for target in range(2, 2 * limit + 1):

            moves = 0


            for i in range(n // 2):

                a = nums[i]
                b = nums[n - 1 - i]

                # 0 move
                if a + b == target:
                    continue

                # 1 move possible
                elif (1 <= target - a <= limit or
                      1 <= target - b <= limit):

                    moves += 1

                # otherwise 2 moves
                else:
                    moves += 2

            ans = min(ans, moves)

        return ans
         